optional_plugins/varianter_cit/avocado_varianter_cit/Cit.py

```diff
@@ -303,15 +303,15 @@ def compute_hamming_distance(self, row):
         return distance
 
     def create_random_row_with_constraints(self):
-        row = []
-        data_matrix = []
-        for parameter in self.data:
-            data_matrix.append(list(range(0, parameter)))
+        """
+        Creates a new test case random row and the row meets the constraints
+        :return: new random row
+        :rtype: list
+        """
+        row = [-1]*len(self.data)
 
-        # delete the forbidden values ​by constraints
-        self.solver.clean_data_matrix(data_matrix)
-        for parameter, possible_values in enumerate(data_matrix):
+        for parameter in random.sample(range(len(self.data)), len(self.data)):
```

Comment:
You are calculating the `len(data)` 3 times here. Maybe we could add a `data_size = len(data)` and use this.

Reply:
Yes, you are right the `data_size` will be better.